src/compiler/context.js

```diff
@@ -0,0 +1,228 @@
+import util from './utils/builder';
+const STATES = {
+  OUTER_SPACE: 'OUTER_SPACE',
+  HTML_TAG: 'HTML_TAG',
+  HTML_BODY: 'HTML_BODY',
+  HTML_ATTRIBUTE: 'HTML_ATTRIBUTE',
+  ESCAPABLE_RAW: 'ESCAPABLE_RAW',
+  TORNADO_TAG: 'TORNADO_TAG',
+  TORNADO_BODY: 'TORNADO_BODY'
+};
+
+let Context = function(results) {
+  // let refCount;
+  let tornadoBodiesPointer;
+  const defaultState = STATES.OUTER_SPACE;
+
+
+
+  let init = function() {
+    // refCount = 0;
+    this.htmlBodies = [];
+    this.tdBodies = [];
+    this.blocks = {};
+    tornadoBodiesPointer = this.tdBodies.length;
+    this.state = defaultState;
+  };
+  let pushFragment = function(fragment, append){
+    if (append) {
+      results.fragments[tornadoBodiesPointer] += fragment;
+    } else {
+      results.fragments[tornadoBodiesPointer] = fragment;
+    }
+  };
+  let pushRenderer = function(renderer, append){
+    if (append) {
+      results.renderers[tornadoBodiesPointer] += renderer;
+    } else {
+      results.renderers[tornadoBodiesPointer] = renderer;
+    }
+  };
+  this.push = function(name, f, r) {
```

Comment:
I love the separation that this creates, but I'm not sure I understand why these functions are a part of `context.js`